packages/rules-engine/src/lib/custom-rule-engine/operations/evaluate.ts

```diff
@@ -0,0 +1,64 @@
+import { logger } from 'nx/src/utils/logger';
+import { CustomRule } from '@/lib/custom-rule-engine/types/custom-rule';
+import { castValue } from '@/lib/custom-rule-engine/helpers/cast-value';
+import { castCompareWithEval } from '../helpers/cast-compare-value';
+
+export const evaluate: CustomRule = ({ rule, context, fieldPath }) => {
+  const { options } = rule;
+  const { operations, compareValue: comparableValue } = options;
+  if (!fieldPath) {
+    logger.warn('Field path is not provided');
+
+    return false;
+  }
+  if (!comparableValue) {
+    logger.warn('Comparable value is not provided');
+
+    return false;
+  }
+
+  const fieldValue = context.entity[fieldPath];
+  const compareValue = castValue(options, fieldValue);
+  const targetValue = castCompareWithEval(options, comparableValue);
+
+  switch (operations) {
+    case 'equal':
```

Comment:
Idea: use functions from lodash, the usage will be looking at lodash docs and using a function from there - could cover a lot of use cases with well tested and mature codebase